flex-templates/java/regional_dlp_de_identification/src/main/java/org/apache/beam/samples/DLPTextToBigQueryStreaming.java

```diff
@@ -0,0 +1,780 @@
+      ContentItem tableItem = ContentItem.newBuilder().setTable(nonEncryptedData).build();
+      this.requestBuilder.setItem(tableItem);
+      DeidentifyContentResponse response = dlpServiceClient.deidentifyContent(this.requestBuilder.build());
+      Table tokenizedData = response.getItem().getTable();
+      numberOfRowsTokenized.update(tokenizedData.getRowsList().size());
+      numberOfBytesTokenized.update(tokenizedData.toByteArray().length);
+      c.output(KV.of(key, tokenizedData));
+    }
+  }
+
+  /**
+   * The {@link TableRowProcessorDoFn} class process tokenized DLP tables and
+   * convert them to
+   * BigQuery Table Row.
+   */
+  public static class TableRowProcessorDoFn extends DoFn<KV<String, Table>, KV<String, TableRow>> {
+
+    @ProcessElement
+    public void processElement(ProcessContext c) {
+      Table tokenizedData = c.element().getValue();
+      List<String> headers = tokenizedData.getHeadersList().stream()
+          .map(fid -> fid.getName())
+          .collect(Collectors.toList());
+      List<Table.Row> outputRows = tokenizedData.getRowsList();
+      if (outputRows.size() > 0) {
+        for (Table.Row outputRow : outputRows) {
+          if (outputRow.getValuesCount() != headers.size()) {
+            throw new IllegalArgumentException(
+                "CSV file's header count must exactly match with data element count");
+          }
+          c.output(
+              KV.of(
+                  c.element().getKey(),
+                  createBqRow(outputRow, headers.toArray(new String[headers.size()]))));
+        }
+      }
+    }
+
+    private static TableRow createBqRow(Table.Row tokenizedValue, String[] headers) {
+      TableRow bqRow = new TableRow();
+      AtomicInteger headerIndex = new AtomicInteger(0);
+      tokenizedValue
+          .getValuesList()
+          .forEach(
+              value -> {
+                String checkedHeaderName = checkHeaderName(headers[headerIndex.getAndIncrement()].toString());
+                bqRow.set(checkedHeaderName, value.getStringValue());
+              });
+      return bqRow;
+    }
+  }
+
+  private static String getJsonSchema(ValueProvider<String> inputSchema) {
+    String []fields = inputSchema.get().split(",");
+    String jsonSchema = "{ \"fields\": [";
+
+    for (int i = 0; i < fields.length; i++) {
+      jsonSchema += String.format("{\"name\": \"%s\", \"type\": \"%s\"}%s", fields[i].split(":")[0].trim(), fields[i].split(":")[1].trim(),
+          i == fields.length - 1 ? "": ",");
+    }
+    jsonSchema += "]}";
+    return jsonSchema;
+  }
+
+  private static String getFileName(ReadableFile file) {
+    String csvFileName = file.getMetadata().resourceId().getFilename().toString();
+    /** taking out .csv extension from file name e.g fileName.csv->fileName */
+    String[] fileKey = csvFileName.split("\\.", 2);
+
+    if (!fileKey[1].equals(ALLOWED_FILE_EXTENSION) || !TABLE_REGEXP.matcher(fileKey[0]).matches()) {
```

Comment:
not needed, but if support differnet types, consider using "matches" for extensions of equals.